fhir/com.b2international.snowowl.fhir.rest/src/com/b2international/snowowl/fhir/rest/FhirValueSetExpandOperationController.java

```diff
@@ -0,0 +1,135 @@
+	 * @param valueSetId
+	 * @return expanded {@link ValueSet}
+	 */
+	@Operation(
+		summary="Expand a value set",
+		description="Expand a value set specified by its logical id."
+	)
+	@ApiResponses({
+		@ApiResponse(responseCode = "200", description = "OK"),
+		@ApiResponse(responseCode = "400", description = "Bad Request"),
+		@ApiResponse(responseCode = "404", description = "Not Found")
+	})
+	@GetMapping(value="/{id:**}/$expand")
+	public Promise<ValueSet> expand(
+			@Parameter(description = "The id of the value set to expand") 
+			@PathVariable("id") 
+			final String id) {
+		return FhirRequests.valueSets().prepareExpand()
+				.setRequest(ExpandValueSetRequest.builder().url(id).build())
+				.buildAsync()
+				.execute(getBus());
+	}
+	
+	/**
+	 * HTTP Get request to expand a value set specified by its URL
+	 * @param url
+	 * @return expanded {@link ValueSet}
+	 */
+	@Operation(
+		summary="Expand a value set",
+		description="Expand a value set specified by its url."
+	)
+	@ApiResponses({
+		@ApiResponse(responseCode = "200", description = "OK"),
+		@ApiResponse(responseCode = "400", description = "Bad request"),
+		@ApiResponse(responseCode = "404", description = "Value set not found")
+	})
+	@GetMapping(value="/$expand")
+	public Promise<ValueSet> expandByURL(
+			@Parameter(description = "Canonical URL of the value set") 
+			@RequestParam(value="url") 
+			final String url) {
+		return FhirRequests.valueSets().prepareExpand()
+				.setRequest(ExpandValueSetRequest.builder().url(url).build())
+				.buildAsync()
+				.execute(getBus());
+	}
+	
+	/**
+	 * HTTP Post request to expand a value set
+	 * @param body - FHIR parameters
+	 * @return expanded {@link ValueSet}
+	 */
+	@Operation(
+		summary="Expand a value set",
+		description="Expand a value set specified by a request body."
+	)
+	@ApiResponses({
+		@ApiResponse(responseCode = "200", description = "OK"),
+		@ApiResponse(responseCode = "400", description = "Bad request"),
+		@ApiResponse(responseCode = "404", description = "Value set not found")
+	})
+	@PostMapping(value="/$expand", consumes = AbstractFhirResourceController.APPLICATION_FHIR_JSON)
+	public Promise<ValueSet> expandBodyRequest(
+			@Parameter(description = "The lookup request parameters")
+			@RequestBody 
+			Parameters.Fhir body) {
+		
+		final ExpandValueSetRequest request = toRequest(body, ExpandValueSetRequest.class);
+		
+		if (request.getUrl() == null && request.getValueSet() == null) {
+			throw new BadRequestException("Both URL and ValueSet parameters are null.", "ExpandValueSetRequest");
+		}
+
+		if (request.getUrl() == null || request.getUrl().getUriValue() == null) {
+			throw new BadRequestException("Expand request URL is not defined.", "ExpandValueSetRequest");
+		}
+		
+		if (request.getUrl() != null && 
+				request.getValueSet() != null && 
+				request.getUrl().getUriValue() != null &&
+				request.getValueSet().getUrl().getUriValue() != null &&
```

Comment:
We don't know whether `request.getValueSet().getUrl()` is non-null at this point 😃

Reply:
Same here, this was the original code in 7.x. Also, we won't support the `valueSet` property of the `$expand` operation, feel free to ignore this for now.